ui/src/components/Modals/OasModal/OasModal.tsx

```diff
@@ -24,11 +24,12 @@ const OasModal = ({ openModal, handleCloseModal }) => {
         console.error(e);
       }
     })();
+    return () => setOasServices([]);
   }, [openModal]);
 
   const onSelectedOASService = async (selectedService) => {
     setSelectedServiceName(selectedService);
-    if(oasServices.length === 0){
+    if(!oasServices || oasServices?.length === 0){
```

Comment:
I think the check of !oasServices is redundant cuz you init this state as an empty array

Reply:
Happened because UI feature flag was true while server feature flag was false. shouldn't happen normally.